src/test/java/org/prebid/server/handler/openrtb2/AuctionHandlerTest.java

```diff
@@ -217,6 +227,37 @@ public void shouldRespondWithBidResponse() {
         verify(httpResponse).end(eq("{}"));
     }
 
+    @Test
+    public void shouldRespondWithCorrectResolvedRequestMediaTypePriceGranularity() throws IOException {
+        // given
+        given(auctionRequestFactory.fromRequest(any(), anyLong()))
+                .willReturn(Future.succeededFuture(givenAuctionContext(identity())));
+
+        final BidRequest resolvedRequest = BidRequest.builder()
+                .ext(mapper.valueToTree(ExtBidRequest.of(ExtRequestPrebid.builder()
+                        .targeting(ExtRequestTargeting.of(null,
+                                ExtMediaTypePriceGranularity.of(mapper.valueToTree(ExtPriceGranularity.of(1,
+                                        singletonList(ExtGranularityRange.of(BigDecimal.TEN, BigDecimal.ONE)))),
+                                        null, null), null, null, null))
+                        .build())))
+                .build();
+        given(exchangeService.holdAuction(any()))
+                .willReturn(Future.succeededFuture(BidResponse.builder()
+                        .ext(mapper.valueToTree(ExtBidResponse.of(ExtResponseDebug.of(null, resolvedRequest),
+                                null, null, null, null)))
```

Comment:
Pls set native to `mapper.createObjectNode()` to check the result.

Reply:
Will do. Left it like this as the reported case was when initial request had no `native` at all, but it would still include `"native": null` and `"xnative": null` fields, even though we have our mapper set to exclude null fields in json.

Reply:
Thanks.